overlord/state/taskrunner.go

```diff
@@ -336,6 +336,14 @@ func (r *TaskRunner) Ensure() {
 			}
 			continue
 		}
+
+		// We may have to wait for dependencies, as further tasks might have handlers again.
+		// Cannot undo.
+		if mustWait(t) {
```

Comment:
This package is very well tested, and in isolation from everything else. Can we please have a new test in one of the existing tables that verify the task runner, ensuring this particular case is taken care of?

Reply:
Ok, I've enhanced the existing tests.
Note that without the fix the tests do not always fail, there is an aspect in the taskrunner that's not fully deterministic. I've run these tests hundreds of times in a loop and without the fix it fails randomly every now and then (sometimes it fails in one of the test configurations I haven't touched directly, that's because the mere addition of a new task t23 affects all of them).

Running the tests in a loop with the fix is consistently passing of course.

I would love a 100% deterministic test but not sure if that's possible? That's also matches my observations from the refresh-hooks branch where I discovered the problem - every now and then bug would actually not manifest itself.